fuzzers/libafl-fuzz/src/main.rs

```diff
@@ -0,0 +1,246 @@
+#![feature(io_error_more)]
+#![deny(clippy::pedantic)]
+#![allow(clippy::unsafe_derive_deserialize)]
+#![allow(clippy::ptr_arg)]
+#![allow(clippy::unnecessary_wraps)]
+#![allow(clippy::module_name_repetitions)]
+#![allow(clippy::similar_names)]
+use std::{collections::HashMap, path::PathBuf};
+mod afl_stats;
+mod feedback;
+use clap::Parser;
+use corpus::{check_autoresume, remove_main_node_file};
+mod corpus;
+mod executor;
+mod fuzzer;
+mod hooks;
+mod utils;
+use fuzzer::run_client;
+use libafl::{
+    events::{CentralizedLauncher, EventConfig},
+    monitors::MultiMonitor,
+    Error,
+};
+use libafl_bolts::{
+    core_affinity::{CoreId, Cores},
+    shmem::{ShMemProvider, UnixShMemProvider},
+};
+use nix::sys::signal::Signal;
+use utils::PowerScheduleCustom;
+
+#[allow(clippy::too_many_lines)]
+fn main() {
+    env_logger::init();
+    let mut opt = Opt::parse();
+    executor::check_binary(&mut opt, SHMEM_ENV_VAR).expect("binary to be valid");
+
+    // Create the shared memory map provider for LLMP
+    let shmem_provider = UnixShMemProvider::new().unwrap();
+
+    // Create our Monitor
+    let monitor = MultiMonitor::new(|s| println!("{s}"));
+    /*     let monitor = MultiMonitor::new(|_s| {}); */
+
+    opt.auto_resume = if opt.auto_resume {
+        true
+    } else {
+        opt.input_dir.as_os_str() == "-"
+    };
+
+    if !opt.output_dir.exists() {
+        std::fs::create_dir(&opt.output_dir).unwrap();
```

Comment:
Just always use `crate_dir` and ignore the result? Else spawning multiple nodes at the same time might be racey

Reply:
This is before we run the  launcher

Reply:
Ppl can spawn multiple instances, especially in afl-fuzz. Also, what's to downside of doing it non-racey?